### PR TITLE
970 add a ci to armonikutils repository

### PR DESCRIPTION
**Pull request by ddiakiteaneo**
- [x] CI added to the repository
- [x]  Test fails for now

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    - name: Generate Version
+      id: genver
+      run: |
+        VERSION=$(docker run --rm -v $(pwd):/repo codacy/git-version /bin/git-version --folder=/repo --release-branch=${{ github.ref_name }}-pre --dev-branch=main)
+        echo "version=$VERSION" >> $GITHUB_OUTPUT
+        echo "VERSION : $VERSION"
+        echo "Version :" >> $GITHUB_STEP_SUMMARY
+        echo "$VERSION" >> $GITHUB_STEP_SUMMARY
+
+ format:
```
Comment (aneojgurhem) [inline .github/workflows/deploy.yml:31]: I'd put formatting job in its own worflow file with a trigger on pull requests as it is done in ArmoniK.Core

Comment (ddiakiteaneo) [reply .github/workflows/deploy.yml]: Done !

File: csharp/tests/ParallelSelectExtTest.cs [hunk 2]
```diff
@@ -154,12 +154,12 @@ private static IEnumerable ParallelSelectLimitCases()
                           useAsync,
                           2,
                           20);
-        yield return Case(blocking,
-                          useAsync,
-                          10,
-                          100);
         if (!blocking)
         {
+          yield return Case(blocking,
```
Comment (aneojgurhem) [inline csharp/tests/ParallelSelectExtTest.cs:159]: This should come from another PR and merge or rebase

**Comment by aneojgurhem** [comment]
I think you can remove some of the commits with a rebase since most of them were tests with the ci

File: csharp/tests/ParallelSelectExtTest.cs (hunk 2 above)
Comment (ddiakiteaneo) [reply csharp/tests/ParallelSelectExtTest.cs:159]: Noted !

Comment (aneojgurhem) [reply csharp/tests/ParallelSelectExtTest.cs:159]: I'm gonna merge it

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    - name: Run tests
+      run: |
+        pwd
+        cd csharp
+        dotnet test --logger "trx;LogFileName=test-results.trx"
+  
+    - name: Test Report
+      uses: dorny/test-reporter@v1
+      if: success() || failure()
```
Comment (lemaitre-aneo) [inline .github/workflows/deploy.yml:80]:
```suggestion
      if: always()
```

File: .github/workflows/code-formatting.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  format:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ github.ref }}
+        fetch-depth: 0
+        submodules: true
```
Comment (lemaitre-aneo) [inline .github/workflows/code-formatting.yml:16]: We do not have submodules. So you can remove this.

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+    - name: Pack the package VERSION
+      run: |
+        cd csharp
+        dotnet pack ArmoniK.Utils.sln -c Release -o /tmp/packages -p:Version=$GENVERSION
+      
+    - name: Push the package
+      run: |
+        cd csharp
+        dotnet nuget push /tmp/packages/ArmoniK.Utils.*.nupkg --api-key ${{ secrets.NUGET_API_KEY }} --source https://api.nuget.org/v3/index.json
```
Comment (lemaitre-aneo) [inline .github/workflows/deploy.yml:58]: You should avoid pushing tests

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (ddiakiteaneo) [reply .github/workflows/deploy.yml]: Done !

File: .github/workflows/code-formatting.yml (hunk 4 above)
Comment (ddiakiteaneo) [reply .github/workflows/code-formatting.yml:16]: Done !

**Review by ngruelaneo — APPROVED** [review]

**Pull request merged**